### PR TITLE
Fix chainId not being passed in for dapp browser getProvider function

### PR DESCRIPTION
**Pull request by jinchung**
Fixes APP-1876

## What changed (plus any additional context for devs)
* Main fix: as part of the network > chainId changes, one of the functions got switched from passing in an object param to passing in the chainId directly.
* Upgraded the rainbow provider package to be caught up with the latest changes - this introduced an API change where `getChain` is now `getChainNativeAsset`

## Screen recordings / screenshots
I forgot to take a recording but I did successfully open up a 55x leveraged position.
Screenshot 2024-09-19 at 9 46 33 PM

## What to test
* Please find repro in original linear ticket to test.
* Also generally test dapp browser functionality with various dapps and transactions.

### CODE REVIEW

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 1]
```diff
@@ -344,9 +343,9 @@ export const handleProviderRequestApp = ({ messenger, data, meta }: { messenger:
     checkRateLimit,
     onSwitchEthereumChainNotSupported,
     onSwitchEthereumChainSupported,
-    getProvider: chainId => getProvider({ chainId: chainId as number }) as unknown as Provider,
+    getProvider: ({ chainId }) => getProvider({ chainId: chainId as number }),
```
Comment (jinchung) [inline src/components/DappBrowser/handleProviderRequest.ts:346]: this is the main fix:  as part of the network > chainId changes, one of the functions got switched from passing in an object param to passing in the chainId directly.

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 2]
```diff
@@ -344,9 +343,9 @@ export const handleProviderRequestApp = ({ messenger, data, meta }: { messenger:
...
     getActiveSession,
-    getChain: chainId => defaultChains[chainId] as Chain,
+    getChainNativeCurrency: chainId => chainsNativeAsset[chainId],
```
Comment (jinchung) [inline src/components/DappBrowser/handleProviderRequest.ts:348]: this is not part of the "main fix" but is a consequence of updating to the latest provider package

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@jin/chainid-appsession/bd98927eff4a915d7260b9b24593a405160eac91.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@jin/chainid-appsession/bd98927eff4a915d7260b9b24593a405160eac91.ipa) for bd98927eff4a915d7260b9b24593a405160eac91

**Review by greg-schrammel — APPROVED** [review]
nice jin is a degen opening x55 leverage positions

**Review by brunobar79 — COMMENTED** [review]
🙏🙏🙏

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good 👍🏽

**Pull request merged**